ethcore/src/private_transactions/mod.rs

```diff
@@ -0,0 +1,907 @@
+// Copyright 2015-2017 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+
+// Parity is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Private transactions module.
+/// Export encryptor module.
+pub mod encryptor;
+/// Export the private_transactions module.
+pub mod private_transactions;
+
+extern crate ethabi;
+
+pub use self::encryptor::{Encryptor, SecretStoreEncryptor, DummyEncryptor};
+pub use self::private_transactions::*;
+
+use std::sync::{Arc, Weak};
+use std::collections::HashMap;
+use ethereum_types::{H128, H256, U256, Address};
+use hash::keccak;
+use rlp::*;
+use parking_lot::{Mutex, RwLock};
+use futures::{self, Future};
+use bytes::Bytes;
+use error::Error as EthcoreError;
+use ethkey::{Signature, recover, public_to_address};
+use executive::{Executive, TransactOptions};
+use executed::{Executed};
+use transaction::{SignedTransaction, Transaction, Action, UnverifiedTransaction};
+use client::{Client, BlockChainClient, ChainNotify, ChainMessageType, BlockId, MiningBlockChainClient};
+use account_provider::AccountProvider;
+use service::ClientIoMessage;
+use error::{PrivateTransactionError};
+use ethcore_miner::transaction_queue::{TransactionDetailsProvider as TransactionQueueDetailsProvider, AccountDetails};
+use miner::MinerService;
+use native_contracts::Private as Contract;
+use trace::{Tracer, VMTracer};
+use rustc_hex::FromHex;
```

Comment:
Seems like a lot of imports, perhaps it's possible to split remainder of this file into separate files?